File: test/anim/disabledconn/Node.cc
#include <omnetpp.h>

using namespace omnetpp;

class Node : public cSimpleModule
{
  public:
    virtual void initialize() override;
    virtual void handleMessage(cMessage *msg) override;
};

Define_Module(Node);

void Node::initialize()
{
    scheduleAt(0, new cMessage("timer"));
}

void Node::handleMessage(cMessage *msg)
{
    if (msg->isSelfMessage()) {
        EV << "Sending...\n";
        scheduleAt(simTime()+0.1, msg);
        send(new cPacket("pkt"), "g$o");
    }
    else {
        EV << "Received " << msg->getName() << "\n";
        delete msg;
    }
}

File: contrib/jsimplemodule/jsamples/src/jsamples/Node.java
package jsamples;

import org.omnetpp.simkernel.JSimpleModule;
import org.omnetpp.simkernel.SimTime;
import org.omnetpp.simkernel.cObjectFactory;
import org.omnetpp.simkernel.cMessage;

public class Node extends JSimpleModule {

    protected void initialize() {
        scheduleAt(new SimTime(1), new cMessage("timer"));
    }

    protected void handleMessage(cMessage msg) {
        if (msg.getClassName().equals("IPHeader")) {
            ev.println("source=" + msg.getField("source"));
            ev.println("destination=" + msg.getField("destination"));
            ev.println("extrainfo=" + msg.getField("extrainfo"));
        }
        msg.delete();

        cMessage pk = cMessage.cast(cObjectFactory.createOne("IPHeader"));
        pk.setField("source", "1");
        pk.setField("destination", "2");
        pk.setField("extrainfo", "bla");
        send(pk, "out");
    }
}
